shared/js/background/amp-protection.js

```diff
@@ -48,10 +79,25 @@ function extractAMPURL (site, url) {
 
     for (const regexPattern of ampSettings.linkFormats) {
         const match = url.match(regexPattern)
+        let needsScheme = false
         if (match && match.length > 1) {
-            const newSite = new Site(match[1].startsWith('http') ? match[1] : `https://${match[1]}`)
+            const targetUrl = match[1]
+            console.log(targetUrl)
+            if (!isHttpUrl(targetUrl)) {
+                try {
+                    // eslint-disable-next-line no-unused-vars
+                    const newUrl = new URL(`https://${targetUrl}`)
```

Comment:
This control flow is a little weird but I can't think of a better way.

I wonder if making it clear in a comment that the throw from the constructor here is important in validating the URL?